commands/uploader.go

```diff
@@ -112,36 +114,34 @@ func newUploadContext(remote string, dryRun bool) *uploadContext {
 	return ctx
 }
 
-func verifyLocks(remote string) (ours, theirs []locking.Lock) {
+func verifyLocks(remote string) (ours, theirs []locking.Lock, st verifyState) {
 	endpoint := getAPIClient().Endpoints.Endpoint("upload", remote)
-
 	state := getVerifyStateFor(endpoint)
-	if state == verifyStateDisabled {
-		return
-	}
 
 	lockClient := newLockClient(remote)
 
 	ours, theirs, err := lockClient.VerifiableLocks(0)
 	if err != nil {
 		if errors.IsNotImplementedError(err) {
 			disableFor(endpoint)
-		} else if !errors.IsAuthError(err) {
-			Print("Remote %q does not support the LFS locking API. Consider disabling it with:", remote)
-			Print("  $ git config 'lfs.%s.locksverify' false", endpoint.Url)
-
-			if state == verifyStateEnabled {
+		} else if state == verifyStateUnknown || state == verifyStateEnabled {
+			if !errors.IsAuthError(err) {
+				Print("Remote %q does not support the LFS locking API. Consider disabling it with:", remote)
+				Print("  $ git config 'lfs.%s.locksverify' false", endpoint.Url)
+
+				if state == verifyStateEnabled {
+					ExitWithError(err)
+				}
+			} else {
 				ExitWithError(err)
 			}
```

Comment:
OK, I think I got what you're saying here in: https://github.com/git-lfs/git-lfs/pull/2168/commits/1dc2b4730fd14dffc12e12f17fa91ca6ca88e0e3.